xsnap/documentation/StaticModuleRecord.md

```diff
@@ -0,0 +1,181 @@
+# Static Module Records
+
+Revised: January 6, 2022
+
+Warning: These notes are preliminary. Omissions and errors are likely. If you encounter problems, please ask for assistance.
+
+## About
+
+A static module record prepares a module descriptor to be loaded and initialized as a new module into a compartment.
+
+> Compartments can use module descriptors directly instead of indirectly thru static module records.
+
+A static module record also provides access to the module bindings.
+
+## The StaticModuleRecord Constructor
+
+### StaticModuleRecord(descriptor)
+
+Returns a static module record object. Currently XS supports two kinds of module descriptor.
+
+### descriptor.source
+
+The `source` module descriptor constructs a static module record from a string that is parsed and compiled as a module. 
+
+	const smr = new StaticModuleRecord({ source: `
+		import x from "mod";
+		export let y = x;
+	` });
+
+Obviously, such module descriptor requires a JavaScript parser at runtime.
+
+### descriptor.bindings & descriptor.initialize
+
+The `bindings` & `initialize` module descriptor constructs a static module record from module bindings and a function to initialize the module.
+
+	const smr = new StaticModuleRecord({ 
+		bindings:[
+			{ import: "x", from: "mod" },
+			{ export: "y" },
+		], 
```

Comment:
I will have to putter to know for sure, but the shim may be able to meet you here. As you point out, we currently build third-party static-module-records from imports, exports, and reexports. I assume there’s an equivalence without loss information that the shim needs.